components/Search/SearchForm.jsx

```diff
@@ -4,44 +4,52 @@ import TextField from 'material-ui/TextField';
 import RaisedButton from 'material-ui/RaisedButton';
 import ActionSearch from 'material-ui/svg-icons/action/search';
 import CircularProgress from 'material-ui/CircularProgress';
-
+import { connect } from 'react-redux';
+import { getSearchResult, setSearchQuery } from '../../actions';
+import { browserHistory } from 'react-router'
 import style from './search.css';
+import { REDUCER_KEY } from '../../reducers';
 
 const loadingStyle = {
   display: 'block',
   margin: '40px auto 0 auto',
 };
 
-const SearchForm = ({ hintText, onSubmit, disabled }) => {
-  let queryValue = '';
-
+const SearchForm = ({ location, dispatchSearch, dispatchSetQuery, hintText, onSubmit, disabled, query }) => {
   const formSubmit = (e) => {
     e.preventDefault();
-    onSubmit(queryValue);
+    browserHistory.push(`/search?q=${query}`); 
+    dispatchSearch(query);
```

Comment:
You can use the react-router router object to do this as well, it may be the more accepted pattern. See TabBar.jsx for an example of using it (importing withRouter and then wrapping your component with it which injects the router as a prop into your component). If you look for Programmatic Navigation in this website (https://github.com/reactjs/react-router/blob/master/upgrade-guides/v2.0.0.md), it goes over the change. I'm not sure if there is a benefit to using it, but I would assume there is a reason they prescribe that method.